Project.toml

```diff
@@ -1,7 +1,7 @@
 name = "MockTableGenerators"
 uuid = "63c704a1-a924-4685-a9dd-2c8225e4c0d3"
 authors = ["Beacon Biosignals Inc."]
-version = "0.2.0"
+version = "0.2.1"
```

Comment:
_maybe_ this is a breaking change as it causes previously working behaviour to now error...but I can formulate examples that would show that those DAGs are ill-specified and ambigous...

Alternatively we keep existing behaviour and throw a `warn`?

Reply:
I think the Channel typing change is not breaking, since the function is documented to give `table_key => record` pairs, and looking at the implementation we create those pairs here, so the only issue is if the key isn't a symbol, which sounds like a bug.

For the `_generate!` change, I'm not sure if we document if it's supposed to be `AbstractVector` already. If not, that does sound breaking, since folks could be using tuples or something which probably wouldn't be a bug and would break.

Reply:
> I think the Channel typing change is not breaking, since the function is documented to give table_key => record pairs

I should clarify that I added that docstring in #11 and I'm just merging these changes into that branch to reduce diffs

Reply:
but `table_key` is documented to return a `Symbol`

Reply:
>For the _generate! change, I'm not sure if we document if it's supposed to be AbstractVector already

Closes we get is the example in the README from what I can tell, but nothing explicit

Reply:
In any case, I don't think a method error is the right guardrail for this after all without any further context....

Reply:
> I think it makes sense to have a fallback function _generate!(callback, rng::AbstractRNG, dag::Pair{<:TableGenerator,<:Any}, deps) method that does a warning and forwards to vector typed one by collecting.

This was a good call...it not only made this non-breaking but also resolves the ambiguity of dealing with flattened DAGs altogether.

so `Generator(A) => Generator(B) => Generator(C)` is equivalent to `Generator(A) => [Generator(B) => [Generator(C)]]`, which makes intuitive sense if you were to draw it out.

Reply:
ah this is interesting, this does seem quite nice.

I was suggesting something a bit different. I was thinking we could use `collect` so that if you had a `tuple` or something (like a tuple of children), then it would make an `Vector` and hit the next method, and if you had something not-collectable (like a single child) it would error.

Here we basically assume if it isn't an `AbstractVector`, then it's a single child, and otherwise it's a collection of children. That basically makes sense to me, but I do worry that someone who passes a non-abstractvector collection type will get weird results.

I'm not totally sure the right tradeoff here, but I kinda think my original suggestion is less breaking (if we assume stuff like tuple-of-childrens were supposed to be OK), whereas this one might be more ergonomic but is only non-breaking if it's true that all collection of childrens must be abstractvectors.

Reply:
> Here we basically assume if it isn't an AbstractVector, then it's a single child, and otherwise it's a collection of children. That basically makes sense to me, but I do worry that someone who passes a non-abstractvector collection type will get weird results.

> I'm not totally sure the right tradeoff here, but I kinda think my original suggestion is less breaking (if we assume stuff like tuple-of-childrens were supposed to be OK), whereas this one might be more ergonomic but is only non-breaking if it's true that all collection of childrens must be abstractvectors.

does [this test](https://github.com/beacon-biosignals/MockTableGenerators.jl/pull/12/files#diff-3b9314a6f9f2d7eec1d0ef69fa76cfabafdbe6d0df923768f9ec32f27a249c63R50) not show tuple-of-children is ok?

Reply:
that is a pretty convincing test but it's not testing what it looks like since without a trailing `,`, the parens don't make a tuple

Reply:
updated the test and code - tuples and general collections/iterables should now work